bcycle-backend/Controllers/ShareController.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Threading.Tasks;
+using bcycle_backend.Controllers.Utils;
+using bcycle_backend.Models;
+using bcycle_backend.Models.Responses.Views;
+using bcycle_backend.Security;
+using bcycle_backend.Services;
+using Microsoft.AspNetCore.Mvc;
+
+namespace bcycle_backend.Controllers
+{
+    [ApiController]
+    [Route("/api/share")]
+    public class ShareController : ControllerBase
+    {
+        private readonly ShareService _shareService;
+        private readonly TripService _tripService;
+
+        public ShareController(ShareService shareService, TripService tripService)
+        {
+            _shareService = shareService;
+            _tripService = tripService;
+        }
+
+        // GET /api/share/group-trip/:sharingId
+        [HttpGet("group-trip/{sharingId:guid}")]
+        public async Task<ActionResult<ResultContainer<GroupTripShareView>>> GetGroupTrip(Guid sharingId)
```

Comment:
1. There are already endpoints for this in `GroupTripController`/`TripController`, please delete the old endpoints
2. `[AllowAnonymous]` is missing to make this endpoint accessible without login

Reply:
The data structure  that this endpoint return is modified to correspond to the schema that web is expecting (thats from where the `ShareView` suffix in the return class name comes from). `GroupTripController`/`TripController` return the trip data in a more raw/unmodified format, to be consumed by the mobile app. `[AllowAnonymous]`  would be required if the controller was annotated by `[Authorize]`. Both GET endpoints are meant to be public, but now i see that i forgot to annotate POST and DELETE endpoints to require authorization  after i moved them here. Gonna fix that.

Reply:
Nothing will use these two endpoints anymore:

```cs
        // GET /api/trips/:guid
        [HttpGet("{guid:guid}")]
        [AllowAnonymous]
        public async Task<ActionResult<ResultContainer<TripResponse>>> GetPublic(Guid guid) {}

        // GET /api/group-trips/:guid
        [AllowAnonymous]
        [HttpGet("{guid:guid}")]
        public async Task<ActionResult<ResultContainer<GroupTripResponse>>> Get(Guid guid) {}
```

those endpoints can be removed.

Reply:
Oh, that's what you meant. I missed them. Done.